### PR TITLE
DDPB-3537: Refactor org CSV upload

### PR DESCRIPTION
**Pull request by acsauk-mojd**
## Purpose
In preparation for the court order work we need to simplify and prepare the org CSV upload process to support multiple sources and ensure a good level of test coverage. This will free us to make changes more freely when the CourtOrder entity structure is finalised.

Fixes DDPB-3537

## Approach
I've followed some of the same patterns put in place in the LayCSVUploader work that @aiesoftware did. Using a DTO structure allows us to centralise some of the validation and data sanitising that was spread over OrgService. Simplifying the upload steps and moving some of the existing logic to the Client model and report utils hopefully makes the flow easier to understand.

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [x] I have added tests to prove my work
- [ ] The product team have approved these changes

### CODE REVIEW

**Review by coderste — APPROVED** [review]

**Review by coderste — DISMISSED** [review]

File: api/src/AppBundle/v2/Registration/Uploader/OrgDeputyshipUploader.php [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                $uploadResults['errors'][] = $message;
+                continue;
+            }
+        }
+
+        $uploadResults['added'] = $this->added;
+        return $uploadResults;
+    }
+
+    private function handleNamedDeputy(OrgDeputyshipDto $dto)
```
Comment (coderste) [inline api/src/AppBundle/v2/Registration/Uploader/OrgDeputyshipUploader.php:88]: Missing docs

File: api/src/AppBundle/v2/Registration/Uploader/OrgDeputyshipUploader.php [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+            $this->em->persist($namedDeputy);
+            $this->em->flush();
+
+            $this->added['named_deputies'][] = $namedDeputy->getId();
+        }
+
+        $this->namedDeputy = $namedDeputy;
+    }
+
+    private function handleOrganisation(OrgDeputyshipDto $dto)
```
Comment (coderste) [inline api/src/AppBundle/v2/Registration/Uploader/OrgDeputyshipUploader.php:121]: Same as above

File: api/src/AppBundle/v2/Registration/Uploader/OrgDeputyshipUploader.php [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+            $this->em->persist($organisation);
+            $this->em->flush();
+
+            $this->currentOrganisation = $organisation;
+
+            $this->added['organisations'][] = $organisation->getId();
+        }
+    }
+
+    private function handleClient(OrgDeputyshipDto $dto)
```
Comment (coderste) [inline api/src/AppBundle/v2/Registration/Uploader/OrgDeputyshipUploader.php:137]: Last one :)

File: api/src/AppBundle/v2/Registration/Uploader/OrgDeputyshipUploader.php [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+     * @return bool
+     */
+    private function clientHasNewNamedDeputy(Client $client, NamedDeputy $namedDeputy): bool
+    {
+        return
+            null === $client->getNamedDeputy() ||
+            $client->getNamedDeputy()->getDeputyNo() !== $namedDeputy->getDeputyNo();
+    }
+
+    private function handleReport(OrgDeputyshipDto $dto)
```
Comment (coderste) [inline api/src/AppBundle/v2/Registration/Uploader/OrgDeputyshipUploader.php:201]: Maybe not :)

**Review by coderste — CHANGES_REQUESTED** [review]
Just some PHPDocs blocks missing other than that it's all good :D 👍🏻

**Review by Gugandeep — COMMENTED** [review]
Latest changes look good to me.

**Review by coderste — APPROVED** [review]
👍🏻

**Review by coderste — APPROVED** [review]

**Pull request merged**